jupyterlab/extension.py

```diff
@@ -204,6 +204,14 @@ def load_jupyter_server_extension(nbapp):
     # Must add before the root server handlers to avoid shadowing.
     web_app.add_handlers('.*$', handlers)
 
+    # If running under JupyterHub, add more metadata.
+    if hasattr(nbapp, 'hub_prefix'):
```

Comment:
This check doesn't appear to be working, at least not in my testing. At the time the server extension is loaded, it doesn't seem like the `hub_prefix` traitlet has been set.

`labhubapp.py` seems to be working correctly.

Reply:
I'm concerned that we won't actually be able to deprecate `LabHubApp`. Without it, I'm not sure that JupyterHub spawners can easily configure the `LabApp` specific traitlets (which is a problem I'm having while testing this).

Reply:
I'm not sure why you saw this state.  I'm seeing it working fine locally...